src/libopensc/card.c

```diff
@@ -334,6 +334,8 @@ int sc_lock(sc_card_t *card)
 				/* invalidate cache */
 				memset(&card->cache, 0, sizeof(card->cache));
 				card->cache.valid = 0;
+				if (card->sm_ctx.ops.open)
```

Comment:
shouldn't this be using `sc_card_sm_check` instead of *always* opening a SM channel?